src/toggle-button/internal.tsx

```diff
@@ -62,7 +64,10 @@ export const InternalToggleButton = React.forwardRef(
         }}
         {...rest}
         ref={ref}
-        nativeButtonAttributes={nativeButtonAttributes}
+        nativeButtonAttributes={{
+          ...nativeButtonAttributes,
+          style: { ...nativeButtonAttributes?.style, ...getToggleButtonStyles(style) } as React.CSSProperties,
```

Comment:
The internal button component accepts the `style` prop already - why do we merge it with the native attributes instead?

Reply:
Well, I would rather do it as you said: pass part of the style using `style`, and the rest using native attributes. Then add a code comment documenting why do we need that second part. This makes the separation explicit and benefits future refactoring once we have a good approach for nested component styling.